### ISSUE TITLE
incorrect trigonometric definite integral

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#816]
Imported from SourceForge on 2024-07-03 11:35:41
Created by **[gfroger](https://sourceforge.net/u/gfroger/)** on 2021-02-22 15:13:57
Original: https://sourceforge.net/p/maxima/bugs/3718

---

I think that the attachments speak for themselves.
And this time, i am using the latest versions of Maxima & wxMaxima.

Attachments:
- https://sourceforge.net/p/maxima/bugs/3718/attachment/Integral_16_3.jpg
- https://sourceforge.net/p/maxima/bugs/3718/attachment/Integral_16_3.wxmx
- https://sourceforge.net/p/maxima/bugs/3718/attachment/Integral_16_3_2.jpg

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:35:42
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-02-22 17:38:57
Original: https://sourceforge.net/p/maxima/bugs/3718/#13d1

---

- **labels**:  --> integrate, defint, trigonometry
- **summary**: Integral wrong computation 2 --> incorrect trigonometric definite integral

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:35:46
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-02-22 17:38:57
Original: https://sourceforge.net/p/maxima/bugs/3718/#293c

---

Hi Gerard, thanks for the bug report. I can reproduce this bug. Please bear in mind that you can help others in the process to fix the problem by supplying a clear statement of what the problem is instead of waiting for them to figure it out. 

In this case the problem is the following. I'm working w/ Maxima 5.44.0 on macOS with SBCL 2.1.0.
```{maxima}
(%i1) e:(tan(x)^2+1)^2/(16*tan(x)^4+12*tan(x)^2+1);
                                    2        2
                                (tan (x) + 1)
(%o1)                     ---------------------------
                                4            2
                          16 tan (x) + 12 tan (x) + 1
(%i2) assume(t>0);
(%o2)                               [t > 0]
(%i3) integrate(e,x,0,t);
(%o3)                                  0
```
The 0 result is incorrect, it should be some positive value.

Note that `trigsimp` yields an expression which Maxima recognizes it cannot integrate. An unevaluated integral is the right result in that case.
```{maxima}
(%i4) trigsimp(e);
                                      1
(%o4)                     --------------------------
                               4            2
                          5 sin (x) + 10 sin (x) + 1
(%i5) integrate(trigsimp(e), x, 0, t);
                        t
                       /
                       [              1
(%o5)                  I  -------------------------- dx
                       ]       4            2
                       /  5 sin (x) + 10 sin (x) + 1
                        0
```

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:35:49
Created by **[gfroger](https://sourceforge.net/u/gfroger/)** on 2021-02-23 23:49:42
Original: https://sourceforge.net/p/maxima/bugs/3718/#f520

---

I am sorry ...
I thought that the wxmx file and the two other snap shots would have been enough.
I shall try to give more explanations the next time ...
But, of course, the farther in the future, the better it will be.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:35:53
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-07-18 19:51:17
Original: https://sourceforge.net/p/maxima/bugs/3718/#ed56

---

I just tried it again with a post-5.45 version and it looks like `integrate(e, x)` returns a valid antiderivative, while the definite integral is incorrect (namely zero). For the record the antiderivate I get is:
```
(-(7*atan2(5*sin(4*x)-10*sin(2*x),5*cos(4*x)-10*cos(2*x)+1)
 -7*atan2(sin(4*x)-10*sin(2*x),cos(4*x)-10*cos(2*x)+5)
 -3*sqrt(5)
   *atan2((4*sqrt(5)*sin(2*x))/(5*sin(2*x)^2+5*cos(2*x)^2
                                            +(4*sqrt(5)-10)*cos(2*x)-4*sqrt(5)
                                            +9),
          (5*sin(2*x)^2+5*cos(2*x)^2-10*cos(2*x)+1)
           /(5*sin(2*x)^2+5*cos(2*x)^2+(4*sqrt(5)-10)*cos(2*x)-4*sqrt(5)+9))
 -3*sqrt(5)
   *atan2((4*sqrt(5)*sin(2*x))/(sin(2*x)^2+cos(2*x)^2+(4*sqrt(5)-10)*cos(2*x)
                                          -4*5^(3/2)+45),
          (sin(2*x)^2+cos(2*x)^2-10*cos(2*x)+5)
           /(sin(2*x)^2+cos(2*x)^2+(4*sqrt(5)-10)*cos(2*x)-4*5^(3/2)+45))
 -32*x)
 /160)
 +(3*atan2(5*sin(4*x)-10*sin(2*x),5*cos(4*x)-10*cos(2*x)+1)
  -3*atan2(sin(4*x)-10*sin(2*x),cos(4*x)-10*cos(2*x)+5)
  -sqrt(5)*atan2((4*sqrt(5)*sin(2*x))/(5*sin(2*x)^2+5*cos(2*x)^2
                                                   +(4*sqrt(5)-10)*cos(2*x)
                                                   -4*sqrt(5)+9),
                 (5*sin(2*x)^2+5*cos(2*x)^2-10*cos(2*x)+1)
                  /(5*sin(2*x)^2+5*cos(2*x)^2+(4*sqrt(5)-10)*cos(2*x)
                                -4*sqrt(5)+9))
  -sqrt(5)*atan2((4*sqrt(5)*sin(2*x))/(sin(2*x)^2+cos(2*x)^2
                                                 +(4*sqrt(5)-10)*cos(2*x)
                                                 -4*5^(3/2)+45),
                 (sin(2*x)^2+cos(2*x)^2-10*cos(2*x)+5)
                  /(sin(2*x)^2+cos(2*x)^2+(4*sqrt(5)-10)*cos(2*x)-4*5^(3/2)
                              +45))-8*x)
  /20
 -(atan2(sin(4*x)-2*sin(2*x),(5*cos(4*x)-10*cos(2*x)+1)/5)
  -atan2(sin(4*x)-10*sin(2*x),cos(4*x)-10*cos(2*x)+5)-x)
  /5
  ```
  This appears to be correct by differentiating and comparing to the original expression.
  
  So given that's apparently correct, it appears the error is in the definite integral code (which is separate from the indefinite integral code).

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:35:56
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2021-08-30 15:14:33
Original: https://sourceforge.net/p/maxima/bugs/3718/#daa6

---

The problem appears to be from antideriv which is called for the definite integral.

~~~
trace(?antideriv);
integrate(e,x,0,t);
1 Enter antideriv [1/(16*tan(x)^4+12*tan(x)^2+1)]
1 Exit  antideriv x/5
1 Enter antideriv [tan(x)^2/(16*tan(x)^4+12*tan(x)^2+1)]
1 Exit  antideriv -x/5
1 Enter antideriv [tan(x)^4/(16*tan(x)^4+12*tan(x)^2+1)]
1 Exit  antideriv x/5
~~~
Don't see how 
~~~
integrate(1/(16*tan(x)^4+12*tan(x)^2+1),x,0,t)
~~~
can possibly return x/5.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:36:03
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-09-14 05:14:45
Original: https://sourceforge.net/p/maxima/bugs/3718/#81b0

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 11:36:07
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-09-14 05:14:45
Original: https://sourceforge.net/p/maxima/bugs/3718/#93d3

---

The immediate problem is fixed by commit 5e3cb33, so I'm closing this report. However, it brings up a related issue, which is that Maxima returns a discontinuous antiderivative for this problem. I'll open a separate ticket for that if there is not one already.
